static/src/javascripts/projects/common/modules/discussion/comment-box.js

```diff
@@ -0,0 +1,589 @@
+            AUTH_COOKIE_INVALID:
+                'Sorry, your comment was not published as you are no longer signed in. Please sign in and try again.',
+            'READ-ONLY-MODE':
+                'Sorry your comment can not currently be published as commenting is undergoing maintenance but will be back shortly. Please try again in a moment.',
+
+            /* Custom error codes */
+            /* CORS blocked by HTTP/1.0 proxy */
+
+            API_CORS_BLOCKED: `Could not post due to your internet settings, which might be
+                 controlled by your provider. Please contact your administrator
+                 or disable any proxy servers or VPNs and try again.`,
+            API_ERROR: `Sorry, there was a problem posting your comment. Please try
+                another browser or network connection.  Reference code `,
+            EMAIL_VERIFIED: `<span class="d-comment-box__error-meta">Sent. Please check your
+                email to verify your email address. Once verified post your
+                comment.</span>`,
+            EMAIL_VERIFIED_FAIL: `We are having technical difficulties. Please try again later or
+                <a href="/send/email" class="js-id-send-validation-email">
+                <strong>resend the verification</strong></a>.`,
+            EMAIL_NOT_VALIDATED: `Please confirm your email address to comment.<br />
+                If you can't find the email, we can
+                <a href="_#" class="js-id-send-validation-email">
+                <strong>resend the verification email</strong></a> to your email
+                address.`,
+        };
+
+        this.setOptions(options);
+
+        mediator.on('module:identity:validation-email:success', () =>
+            this.verificationEmailSuccess()
+        );
+
+        mediator.on('module:identity:validation-email:fail', () =>
+            this.verificationEmailFail()
+        );
+    }
+
+    onboardingPreviewSuccess(comment: commentType, resp: Object): void {
+        this.getElem('onboarding-preview-body').innerHTML = resp.commentBody;
+    }
+
+    getDiscussionId(): string {
+        return (
+            this.options.discussionId ||
+            this.elem
+                .getAttribute('data-discussion-key')
+                .replace('discussion', '')
+        );
+    }
+
+    setFormState(disabled?: boolean = false): void {
+        const commentBody = this.getElem('body');
+        const submitButton = this.getElem('submit');
+
+        if (disabled || commentBody.value.length === 0) {
+            submitButton.setAttribute('disabled', 'disabled');
+        } else {
+            submitButton.removeAttribute('disabled');
+        }
+    }
+
+    setExpanded(): void {
+        this.setState('expanded');
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getUserData() {
```

Comment:
Sorry, just spotted a missing return value here (and another one below in `cleanErrors`)